src/gepa/strategies/batch_sampler.py

```diff
@@ -38,12 +48,12 @@ def _update_shuffled(self, trainset_size: int):
                 self.shuffled_ids.append(selected_id)
                 self.id_freqs[selected_id] += 1
 
-    def next_minibatch_indices(self, trainset_size: int, iteration: int) -> list[int]:
-        base_idx = iteration * self.minibatch_size
+    def next_minibatch_ids(self, loader: DataLoader[DataId, DataInst], state: GEPAState) -> list[DataId]:
```

Comment:
The logic of both _update_shuffled and next_minibatch_ids assumes that the length of the trainset is fixed. I don't think this will work well when the trainset expands (the fix should be simple, just need to update shuffled ids every time next minibatch_ids is called, so that it continues to reflect the correct training dataset size).